### ISSUE TITLE
Support ES6

### ISSUE THREAD

**Issue by redak105** [reactjs/react-docgen#147]
Hello,
can I ask about support ES6 fat arrows function?

```
_renderContent = (color: string) => { 
  return ( 
     <View style={{backgroundColor: color}} \>
  );
};
```

Thank you,
redak105

**Comment by fkling** [comment]
react-docgen completely supports ES6. However it doesn't support many of the newer experimental proposals, such as [public class fields](https://github.com/tc39/proposal-class-public-fields), which is what you might be using? Can you please clarify your post?

Also see #137 which would solve that problem.

**Comment by redak105** [comment]
Thanks for your reply.
I'm specifically interested in fat arrows functions. Are these supported?

**Comment by fkling** [comment]
If by support you mean "can react-docgen parse code that uses arrow functions", the answer is yes.

**Comment by redak105** [comment]
Sorry for my late response. 
I have this testing class. I run react-docgen. But I didn't get show method in my output.

`/**
 * Test class
 */
export default class TestClass extends Component {

	/**
	 * Constructor
	 * @property props
	 */
	constructor(props) {
		super(props);

		this.opacity = 0;
	};

	/**
	 * show row
	 */
	show = () => {
		opacity = 1;
	};

	/**
	 * Render
	 */
	render() {
		return (
			<View style={[styles.viewRow, {opacity}]}>
        <Text style={styles.textVideoName}>
          {this.props.title}
        </Text>
			</View>
		);
	}
}

TestClass.propTypes = {
	/**
	 * title
	 * @type {String}
	 */
	title: PropTypes.string.isRequired,
};
`

**Comment by DanielRamosAcosta** [comment]
@fkling I'm having errors with arrow functions in class methods (as seen [here](http://stackoverflow.com/a/31362350)) as the render method? I have a component like this:
```jsx
class Foo extends Component {
  (...)

  render = () =>
    <div>
      Foo!
    </div>
}
```

But it throws to me `No suitable component definition found.`... is there any solution to this? Maybe a handler?

**Comment by fkling** [comment]
That's what I meant when I mentioned in an earlier post

> react-docgen completely supports ES6. However it doesn't support many of the newer experimental proposals, such as public class fields, which is what you might be using?

This is an example of a public class field.

However, since then [we have released `v3.0.0-beta`](https://github.com/reactjs/react-docgen/releases/tag/v3.0.0-beta1) which uses Babylon7, which should be able to handle this case!

**Comment by DanielRamosAcosta** [comment]
I'll give it a shot, thanks for your work!
